packages/steps/src/BuildConfigParser.ts

```diff
@@ -31,50 +31,77 @@ import { BuildWorkflowValidator } from './BuildWorkflowValidator.js';
+    {
+      configPath,
+      externalFunctions,
+      externalFunctionGroups,
+    }: {
+      configPath: string;
+      externalFunctions?: BuildFunction[];
+      externalFunctionGroups?: BuildFunctionGroup[];
+    }
   ) {
     this.validateExternalFunctions(externalFunctions);
 
     this.configPath = configPath;
     this.externalFunctions = externalFunctions;
+    this.externalFunctionGroups = externalFunctionGroups;
   }
 
   public async parseAsync(): Promise<BuildWorkflow> {
     const config = await readAndValidateBuildConfigAsync(this.configPath, {
       externalFunctionIds: this.getExternalFunctionFullIds(),
+      externalFunctionGroupsIds: this.getExternalFunctionGroupFullIds(),
     });
     const configBuildFunctions = this.createBuildFunctionsFromConfig(config.functions);
     const buildFunctions = this.mergeBuildFunctionsWithExternal(
       configBuildFunctions,
       this.externalFunctions
     );
-    const buildSteps = config.build.steps.map((stepConfig) =>
-      this.createBuildStepFromConfig(stepConfig, buildFunctions)
+    const buildFunctionGroups = createBuildFunctionGroupByIdMapping(
+      this.externalFunctionGroups ?? []
     );
+    const buildSteps: BuildStep[] = [];
+    for (const stepConfig of config.build.steps) {
+      buildSteps.push(
+        ...this.createBuildStepFromConfig(stepConfig, buildFunctions, buildFunctionGroups)
+      );
+    }
     const workflow = new BuildWorkflow(this.ctx, { buildSteps, buildFunctions });
     await new BuildWorkflowValidator(workflow).validateAsync();
     return workflow;
   }
 
   private createBuildStepFromConfig(
     buildStepConfig: BuildStepConfig,
-    buildFunctions: BuildFunctionById
-  ): BuildStep {
+    buildFunctions: BuildFunctionById,
+    buildFunctionGroups: BuildFunctionGroupById
+  ): BuildStep[] {
     if (isBuildStepCommandRun(buildStepConfig)) {
-      return this.createBuildStepFromBuildStepCommandRun(buildStepConfig);
+      return [this.createBuildStepFromBuildStepCommandRun(buildStepConfig)];
     } else if (isBuildStepBareCommandRun(buildStepConfig)) {
-      return this.createBuildStepFromBuildStepBareCommandRun(buildStepConfig);
-    } else if (isBuildStepBareFunctionCall(buildStepConfig)) {
-      return this.createBuildStepFromBuildStepBareFunctionCall(buildFunctions, buildStepConfig);
+      return [this.createBuildStepFromBuildStepBareCommandRun(buildStepConfig)];
+    } else if (isBuildStepBareFunctionOrFunctionGroupCall(buildStepConfig)) {
+      return this.createBuildStepsFromBuildStepFunctionOrBuildStepFunctionGroupCall(
+        buildFunctions,
+        buildFunctionGroups,
+        buildStepConfig
+      );
     } else if (buildStepConfig !== null) {
-      return this.createBuildStepFromBuildStepFunctionCall(buildFunctions, buildStepConfig);
+      return [this.createBuildStepFromBuildStepFunctionCall(buildFunctions, buildStepConfig)];
```

Comment:
Should we allow inputs to function groups?

Reply:
Maybe in the future, in this PR I would keep it as concise as possible